app/assets/javascripts/preferences/panes/two-factor-auth/AuthAppInfoPopup.tsx

```diff
@@ -0,0 +1,44 @@
+import { IconButton } from '@/components/IconButton';
+import { FunctionComponent } from 'preact';
+import { useState, useRef, useEffect } from 'react';
+
+/**
+ * AuthAppInfoPopup is an info icon that shows a tooltip when clicked
+ * Tooltip is dismissible by clicking outside
+ *
+ * Note: it can be generalized but more use cases are required
+ * @returns
+ */
+export const AuthAppInfoTooltip: FunctionComponent = () => {
+  const [shown, setShow] = useState(false);
+  const ref = useRef(null);
+
+  useEffect(() => {
+    const dismiss = () => setShow(false);
+    document.addEventListener('mousedown', dismiss);
+    return () => {
+      document.removeEventListener('mousedown', dismiss);
+    };
+  }, [ref]);
+
+  return (
+    <div className="relative">
+      <IconButton
+        icon="info"
+        className="mt-1"
+        onClick={() => {
+          setShow(!shown);
```

Comment:
For buttons that control visibility of another element, we use ReachUI's Disclosure https://reach.tech/disclosure. This ensures that we handle accessibility in the best possible way with a quick implementation

Reply:
Done in 311f50d00cef43129e5fa08819e9987a3625ee87.